tensorboard/data_compat.py

```diff
@@ -81,19 +81,45 @@ def make_summary(tag, metadata, data):
 
 
 def _migrate_histogram_value(value):
-    histogram_value = value.histo
-    bucket_lefts = [histogram_value.min] + histogram_value.bucket_limit[:-1]
-    bucket_rights = histogram_value.bucket_limit[:-1] + [histogram_value.max]
-    bucket_counts = histogram_value.bucket
-    buckets = np.array(
-        [bucket_lefts, bucket_rights, bucket_counts], dtype=np.float32
-    ).transpose()
+    """Convert `old-style` histogram value to `new-style`.
 
+    The "old-style" format can have outermost bucket limits of -DBL_MAX and
+    DBL_MAX, which are problematic for visualization. We replace those here
+    with the actual min and max values seen in the input data, but then in
+    order to avoid introducing "backwards" buckets (where left edge > right
+    edge), we first must drop all empty buckets on the left and right ends.
+    """
     summary_metadata = histogram_metadata.create_summary_metadata(
         display_name=value.metadata.display_name or value.tag,
         description=value.metadata.summary_description,
     )
 
+    histogram_value = value.histo
+    bucket_counts = histogram_value.bucket
+    # Find the indices of the leftmost and rightmost non-empty buckets.
+    n = len(bucket_counts)
+    start = next((i for i in range(n) if bucket_counts[i] > 0), n)
+    end = next((i for i in reversed(range(n)) if bucket_counts[i] > 0), -1)
+    if start > end:
+        # If all input buckets were empty, treat it as a zero-bucket
+        # new-style histogram.
+        return make_summary(
```

Comment:
Ah right, I forgot that we need to wrap this in `make_summary()` in two places if we do an early return here. Alternatively we could do something like:

```py
if start > end:
    buckets = np.zeros([0, 3], dtype=np.float32)
else:
    # rest of normal codepath goes here
    buckets = np.array([...], dtype=np.float32).transpose()
return make_summary(value.tag, summary_metadata, buckets)
```

Up to you!

Reply:
Changed the sequence to avoid moving the code blocks around.